MobiFlight/Joysticks/Joystick.cs

```diff
@@ -216,32 +246,33 @@ public List<ListItem<IBaseDevice>> GetAvailableDevicesAsListItems()
 
         private List<JoystickDevice> GetButtonsSorted()
         {
-            var buttons = Buttons.ToArray().ToList();
-            buttons.Sort(SortByPositionInDefintion);
-            return Buttons;
+            return Buttons.OrderBy(button => GetIndexForJoystickDevice(button)).ToList();
         }
 
         private List<JoystickDevice> GetAxisSorted()
         {
-            var axes = Axes.ToArray().ToList();
-            Axes.Sort(SortByPositionInDefintion);
-
-            return Axes;
+            return Axes.OrderBy(axis => GetIndexForJoystickDevice(axis)).ToList();
         }
 
-        public int GetIndexForKey(string key)
+        public virtual int GetIndexForKey(string key)
         {
-            return Definition?.Inputs?.FindIndex(input => input.Name == key) ?? 0;
+            // -2 = definition not found / no inputs in definition
+            // -1 = index not found
+            // 0 or greater: valid index
+            return Definition?.Inputs?.FindIndex(input => input.Name == key) ?? -2;
         }
 
-        int SortByPositionInDefintion(JoystickDevice b1, JoystickDevice b2)
+        public int GetIndexForJoystickDevice(JoystickDevice button)
         {
-            if (GetIndexForKey(b1.Name) == GetIndexForKey(b2.Name)) return 0;
-            if (GetIndexForKey(b1.Name) > GetIndexForKey(b2.Name)) return 1;
-            return -1;
+            var index = GetIndexForKey(button.Name);
+            // Ensure items in definition appear first
+            if(index < 0)
+            {
+                index = Int32.MaxValue; // force undefined stuff to the end of the list
```

Comment:
The inputs most relevant to MobiFlight users are probably spelled out in the definition. Entries not in the definition are likely to be less relevant, so let us move them after the named ones.